### ISSUE TITLE
Lexer: Multiline comment in an array does not compile

### ISSUE THREAD

**Issue by seiyria** [jashkenas/coffeescript#3735]
This code will not compile:

``` coffee
class LightsFromTheStars extends Spell
  name: "Lights from the Stars"
  @element = LightsFromTheStars::element = Spell::Element.buff
  @tiers = LightsFromTheStars::tiers = [
    ###
      * This spell buffs the int and wis of your allies.
      *
      * @name Lights from the Stars
      * @requirement {class} Bard
      * @requirement {mp} 300
      * @requirement {level} 10
      * @effect +[caster.int/4] int
      * @effect +[caster.wis/4] wis
      * @duration 3 rounds
      * @category Bard
      * @package Spells
    ###
    {name: "Lights from the Stars", spellPower: 1, cost: 300, class: "Bard", level: 10}
  ]
```

Is this intentional?

**Comment by GeoffreyBooth** [comment]
It only works if you remove the braces because then the comment is interpreted as _inside_ the object, i.e.:

```js
tiers = [
  {

    /*
      * This spell buffs the int and wis of your allies.
      *
      * @name Lights from the Stars
      * @requirement {class} Bard
      * @requirement {mp} 300
      * @requirement {level} 10
      * @effect +[caster.int/4] int
      * @effect +[caster.wis/4] wis
      * @duration 3 rounds
      * @category Bard
      * @package Spells
     */
    name: "Lights from the Stars",
    spellPower: 1,
    cost: 300,
    "class": "Bard",
    level: 10
  }
];
```

And multiline comments within objects are already supported.

Minimal example:

```coffee
arr = [
  ###
    Comment
  ###
  1
]
```
___
```bash
[stdin]:4:6: error: unexpected newline
  ###
     ^
```

**Comment by GeoffreyBooth** [comment]
Fixed via #4572.
